libs/domains/observability/feature/src/lib/service-overview/local-chart/local-chart.tsx

```diff
@@ -95,27 +95,61 @@ function ChartContent({
     useServiceOverviewContext()
   const [onHoverHideTooltip, setOnHoverHideTooltip] = useState(false)
 
-  function getXDomain() {
-    return xDomain ?? [Number(startTimestamp) * 1000, Number(endTimestamp) * 1000]
+  // Use the zoomable chart hook
+  const {
+    zoomState,
+    isCtrlPressed,
+    handleChartDoubleClick,
+    handleMouseDown,
+    handleMouseMove,
+    handleMouseUp,
+    handleMouseLeave,
+    getXDomain: getZoomXDomain,
+  } = useZoomableChart()
+
+  function getXDomain(): [number | string, number | string] {
+    const defaultDomain: [number | string, number | string] = xDomain ?? [
+      Number(startTimestamp) * 1000,
+      Number(endTimestamp) * 1000,
+    ]
+    return getZoomXDomain(defaultDomain)
   }
 
   const xAxisConfig = createXAxisConfig(Number(startTimestamp), Number(endTimestamp))
 
   return (
-    <div className="flex h-full">
+    <div className="flex h-full" style={{ position: 'relative' }}>
```

Comment:
I think you can directly add `relative` like a class name instead of using jsx style